### PR TITLE
Update Disks logic

### PR DESCRIPTION
**Pull request by cassidyjames**
This is an upstreaming of changes from @mmstick and the pop fork, which gets the installer building again and sets up a lot of the logic needed for the partitioning screen.

### CODE REVIEW

File: src/Objects/Mount.vala [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    }
+
+    public bool is_valid_root_mount () {
+        return filesystem != Distinst.FileSystemType.FAT16
+            && filesystem != Distinst.FileSystemType.FAT32
+            && filesystem != Distinst.FileSystemType.NTFS;
+    }
+
+    public bool is_lvm () {
+        return (this.flags & LVM) != 0;
```
Comment (tintou) [inline src/Objects/Mount.vala]: Vala has a syntax for checking flags:
`return LVM in flags;`

File: src/Objects/Mount.vala [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    }
+
+    public bool should_format () {
+        return (this.flags & FORMAT) != 0;
```
Comment (tintou) [inline src/Objects/Mount.vala]: `return FORMAT in flags`

File: src/Views/DiskView.vala [hunk 3]
```diff
@@ -144,7 +139,7 @@ public class Installer.DiskView : AbstractInstallerView {
                 label = model;
             }
 
-            string path = (owned) path_builder.str;
+            string path = Utils.string_from_utf8(disk.get_device_path ());
```
Comment (tintou) [inline src/Views/DiskView.vala]: spacing here

File: src/Views/ProgressView.vala [hunk 4]
```diff
@@ -297,19 +327,102 @@ public class ProgressView : AbstractInstallerView {
...
+
+                if (m.mount_point == "/boot/efi") {
+                    if (m.is_valid_boot_mount ()) {
+                        if (m.should_format ()) {
+                            partition.format_with (m.filesystem);
+                        }
+
+                        partition.set_mount (m.mount_point);
+                        Distinst.PartitionFlag[] flags = { Distinst.PartitionFlag.ESP };
+                        partition.set_flags (flags);
```
Comment (tintou) [inline src/Views/ProgressView.vala]: Use `partition.set_flags ({ Distinst.PartitionFlag.ESP });` as done just after

**Comment by cassidyjames** [comment]
@tintou thanks for the quick review! I've addressed your comments.

**Comment by jackpot51** [comment]
:tada:

**Pull request merged**